src/ui/public/notify/notify.js

```diff
@@ -70,18 +78,53 @@ if (!!metadata.kbnIndex) {
   });
 }
 
+let bannerId;
```

Comment:
I assume these are placed here so that when multiple notifiers are created, like in the tests, the previous banner and timeouts can be clears correctly? Could we instead pass the `$rootScope` to `applyConfig()` and add teardown listener in a `$rootScope.$on('$destroy')` listener so this intention is more clear and we aren't using module scope to store state?

Reply:
These variables just allow us to remove the banner when the user clicks its "Close" button or when its lifetime elapses. This logic used to live inside of the Notifier class but now it's executed once in a run block so I don't think the concern you brought up applies any more -- correct me if I'm wrong.